polaris-core/src/main/java/org/apache/polaris/core/persistence/PolarisMetaStoreManager.java

```diff
@@ -285,7 +285,11 @@ DropEntityResult dropEntityIfExists(
    * @param entityId the id of the entity to load
    */
   @Nonnull
-  EntityResult loadEntity(@Nonnull PolarisCallContext callCtx, long entityCatalogId, long entityId);
+  EntityResult loadEntity(
+      @Nonnull PolarisCallContext callCtx,
+      long entityCatalogId,
+      long entityId,
+      @Nonnull PolarisEntityType entityType);
```

Comment:
We should document in the javadoc whether this is intended to only be a "hint" or whether there's expectations that the type is enforced strictly. In particular, since we've defined `entityId` to need to be unique within the realm, if the underlying impl happens be able to look up the entity without checking `entityType` we should mention if it's safe for that impl to simpy ignore this param.

Due to the nature of this, I would prefer to let it be `entityTypeHint` and allow impls to ignore it, and distinguish the intended behavior from the strict entityType in `readEntityByName`.

Reply:
Wait, if it can be ignored why are we even passing it in?

Reply:
Is there a situation where we _wouldn't_ want to enforce that entityType matches, even if the impl technically could look up an entity without it?

Reply:
In general, from my POV, it is perfectly fine for an API to have redundant inputs. Those can be used for validation as well as for optimization.

As for validation, the current behaviour expectation is to "not find" the object if the given type parameter does not match actual type in storage. I will update javadoc about that if we agree on adding the new parameter.